### ISSUE TITLE
Bug in HTML generated of the return value of \retval commands

### ISSUE THREAD

**Issue by maribu** [doxygen/doxygen#7499]
### Steps to Reproduce

Run Doxygen on the following input

```
/**
 * @brief Does foo
 * @retval `-1`    on error
 * @retval `0`     on success
 */
int foo(void) {
    return  has_error() ? -1 : 0;
}
```

### Expected Result

In the return value table the values `-1` and `0` are formatted as source code. (E.g. `<tt>-1</tt>` and `<tt>0</tt>`.)

### Actual Result

`&lt;tt&gt;-1&lt;/tt&gt;` and `&lt;tt&gt;0&lt;/tt&gt;` is generated instead.

### Affected Version

Doxygen 1.8.17

**Comment by albert-github** [comment]
I don't think this is a bug. The return value, like the parameter name with the `\param` statement, are to be show as just the name / value.
When one wants to show, all, e.g. return values in a certain style, one should do this, for the HTML output, in an `EXTRA_HTML_STYLESHEET` where one redefines the elements inside the:
```
.retval .paramname { 
 ...
}
```

**Comment by maribu** [comment]
> I don't think this is a bug. The return value, like the parameter name with the `\param` statement, are to be show as just the name / value.

If the intended behavior is to indeed just show the name / value, then e.g. <code>&#96;-1&#96;</code> should generate `&#96;-1&#96;` as HTML output, so that &#96;-1&#96; is displayed.

The current behavior is that the markdown is indeed parsed, corresponding HTML tags are generated (e.g. <code>&#96;-1&#96;</code> is transformed to `<tt>-1</tt>`), but then the generated HTML is escaped to `&lt;tt&gt;0&lt;/tt&gt;`. So this is also a bug when support for using markdown to style values and names is explicitly not supported.

But: Why should the use of markdown for styling @param names and values be disabled? There might be use cases were some return values are highlighted compared to others, which is impossible by using CSS that would affect all param name cells.

**Comment by albert-github** [comment]
To overcome the "literal" backtick problem one can use ``"`-1`"``. 

@doxygen what to do with more styling possibilities for return values / parameter names etc?

**Comment by doxygen** [comment]
@albert-github Some extra styling options maybe useful, but I wonder where it will end. Next request may be to support hyperlinked return values or even images...

**Comment by albert-github** [comment]
@doxygen I do agree (and for the hyperlinked return values I could even imagine an use case, i.e. when the return value an enumaration value is).
